assets/js/components/dashboard-sharing/ModuleRecoveryAlert.js

```diff
@@ -79,15 +81,18 @@ export default function ModuleRecoveryAlert() {
 		[]
 	);
 
-	const handleRecoverModules = useCallback( () => {
+	const handleRecoverModules = useCallback( async () => {
 		setRecoveringModules( true );
+
 		const modulesToRecover = Object.keys( checkboxes ).filter(
 			( module ) => checkboxes[ module ]
 		);
-		recoverModules( modulesToRecover ).finally( () => {
-			setRecoveringModules( false );
-			setCheckboxes( null );
-		} );
+		await recoverModules( modulesToRecover );
+
+		setRecoveringModules( false );
+		setCheckboxes( null );
+
+		return { dismissOnCTAClick: false };
```

Comment:
If we want to make this a bit more explicit, I think `dismissCTA` would be ok here, but including `OnClick` is unnecessary detail. There isn't anything particularly special about clicks here except that is the prop this function will be passed to. Let's avoid coupling with clicks where it isn't needed.

Reply:
Well the new flag is defined and used solely in the `onCTAClick` handler - so perhaps the name of this flag is fine? I was being specific mainly because we already have `dismiss` as a prop in `BannerNotification` for the actual dismiss link/CTA. `dismissCTA` could also sound confusing as yet another dismiss CTA link / button instead of just a flag that would prevent the default dismissal of a BannerNotification. Any other name suggestions perhaps? e.g `preventCTADismissal`?

Reply:
It's probably fine as is; it sticks out to me as a leaked detail though because the function here is `handleRecoverModules` rather than `handleOnCTAClick` where `handleRecoverModules` could be bound to other events than a click. In that context, I would opt for something a bit more generic for the response. It sounds like you're thinking of the name as a standalone concept in the component where it could be confused with other props. I'm actually less concerned with the name where it's used, and thinking about it more as an API of the function. So the name you have now would be fine (i.e. as an alias when destructuring), or left on the object it could be a bit more self documenting `onCTAResult.dismissCTA` that kind of thing.

Again, it doesn't matter that much right now because there is only the one use right now and everything else that wasn't using it is unaffected by this change, just sharing my perspective 👍